### ISSUE TITLE
Stripe modal & validation issue

### ISSUE THREAD

**Issue by trilame** [woocommerce/woocommerce-gateway-stripe#391]
Hello,
I'm facing an issue with the stripe modal feature. Basically, some validation checks (postcode format, phone number format, etc.) are triggered after after the Stripe modal popup is filled.
So, we click pay, put card information, validate (it looks like everything went well in the modal), but we are back on the checkout page, with errors.
I tested it with the last version but as well with 4.0 beta.

It is exactly what is described here:
https://github.com/woocommerce/woocommerce-gateway-stripe/issues/161
https://github.com/woocommerce/woocommerce-gateway-stripe/issues/243

It is supposed to be fixed in 4.0, but unfortunately, I'm still facing it with the last beta...

#### What I expected

Woocommerce validations to be runned before showing the Stripe modal page, not after.
#### What happened instead

If there are errors in field post code (for example, text instead of numbers for country France), we can still access to the Stripe modal.
After validation of the strip modal, we are back on the checkout page with the postcode error.

#### Steps to reproduce the issue

- Have any theme (I also tested with Twenty Sixteen)
- Proceed to checkout
- Put something wrong in post code or other field, e.g. "qwerty" in Post code with country "France"
- Try to access to Stripe modal popup

WC: 3.2.1
WP: 4.8.2
Woocommerce stripe gateway: last commit [here](https://github.com/woocommerce/woocommerce-gateway-stripe/commit/e0ac1c3c28e8718810e5c2ffd0ae84d411e415a4)

Thank you :) !

**Comment by roykho** [comment]
Did you test this on mobile or desktop?

**Comment by trilame** [comment]
On both

**Comment by roykho** [comment]
It works on desktop only. Does it validate if you left it empty?

**Comment by roykho** [comment]
Ok I think this validation is specific to format of the postcode. It validates if input was entered but not necessarily the format. This will be fixed when 4.0 is released. Thanks for reporting.

**Comment by trilame** [comment]
Hello @roykho 
I did some tests on desktop only.

If I left empty, the behavior is as expected (doesn't show the modal but a warning message).

I also tested on other fields, and i have the same problem with **"Billing phone"**
If I leave Billing phone empty, it works as expected (doesn't show the modal but a warning message).

If it can help: my wordpress is in French, my theme still in english.
- When I do the test with wrong billing phone format, I have the error in French
- If I left empty, I have the error message in English

Regards
